File: search/linear_search/java/LinearSearchTest.java
import static org.junit.Assert.*;

import org.junit.Test;

public class LinearSearchTest {

	@Test
	public void test() {
		int[] array = {5,4,3,2,1};
		assert(LinearSearch.search(array,2) == 3);
		assert(LinearSearch.search(array,6) == -1);
	}

}

File: search/linear_search/javascript/linear_search_test.js
// use jasmine to run this file
// https://jasmine.github.io
//
// npm install -g jasmine
// jasmine <this_file>

const linearSearch = require('./linear_search');

describe("Linear Search", () => {

  it('search in small array', () => {

    [
      [ [1, 2, 3, 4], 2, 1],
      [ [1, 2, 3, 4], 4, 3],
      [ [1, 2, 3, 4], 5, -1],
      [ [1, 3, 4, 5, 6], 2, -1 ]
    ].forEach( test => {
      let ts = process.hrtime()
      expect(linearSearch(test[0], test[1])).toBe(test[2]);
      ts = process.hrtime(ts);
      console.log(`Search took: ${ts[0] * 1e9 + ts[1] } ns`);
    });
  })

});
